org.civicrm.reqangular/src/common/services/api/contract.js

```diff
@@ -0,0 +1,38 @@
+/* eslint-env amd */
+
+define([
+  'common/modules/apis',
+  'common/services/api'
+], function (apis, _) {
+  'use strict';
+
+  apis.factory('api.contract', ['$log', 'api', function ($log, api) {
+    $log.debug('api.contract');
+
+    return api.extend({
+
+      /**
+       * This method returns all the contracts.
+       *
+       * It chains an additional call to the `getfulldetails` endpoint to also return
+       * full details on contracts
+       *
+       * @param  {Object} params  matches the api endpoint params (ex. contact_id)
+       * @return {Promise}
+       */
+      all: function (params) {
+        $log.debug('ContractAPI.all');
+
+        params = params || {};
+
+        params['api.HRJobContract.getfulldetails'] = {'jobcontract_id': '$value.id'};
```

Comment:
@deb1990 @igorpavlov from the performance point of view, this is fine. Internally, the `getFullDetails` API uses only 2 queries to get everything (one for all the information and another one just for the leave data, because it's a 1-N relationship). I didn't do any actual benchmark on this, but on the scenario specific for this ticket (loading the contracts of a single contact), the difference in performance of both alternatives is negligible. 

Now, about this
> Now as this is a generic API we should send which field we want to fetch from parameters. Unless any specific field is mentioned all the data will be fetched.

`getFullDetails` is a custom API and it doesn't support specifying which fields will be returned.